From 67937d44f0a2192faee7351328612299b9e96acf Mon Sep 17 00:00:00 2001
From: Frediano Ziglio <freddy77@gmail.com>
Date: Mon, 12 Apr 2021 21:15:11 +0100
Subject: [PATCH] Do not fail using CMake if gperf is not present but not
 needed

gperf is needed only if you need to generate some files,
not always.

Signed-off-by: Frediano Ziglio <freddy77@gmail.com>
---
 CMakeLists.txt         | 3 ---
 src/tds/CMakeLists.txt | 3 +++
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e3840ae5d..4dc9d7b66 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,9 +81,6 @@ include(CheckPrototypeDefinition)
 
 find_package(Perl)
 find_program(GPERF NAMES gperf)
-if (NOT GPERF)
-    message(FATAL_ERROR "gperf not found!")
-endif()
 
 macro(config_write str)
 	file(APPEND "${CMAKE_BINARY_DIR}/include/config.h.in" "${str}")
diff --git a/src/tds/CMakeLists.txt b/src/tds/CMakeLists.txt
index c2e52a410..134a30cb8 100644
--- a/src/tds/CMakeLists.txt
+++ b/src/tds/CMakeLists.txt
@@ -1,6 +1,9 @@
 add_subdirectory(unittests)
 
 if(PERL_FOUND AND NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/tds_willconvert.h")
+	if (NOT GPERF)
+	    message(FATAL_ERROR "gperf not found!")
+	endif()
 	# get name of null device
 	set(NULLNAME "/dev/null")
 	if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
